include/seqan3/io/stream/concept.hpp

```diff
@@ -58,17 +58,22 @@ namespace seqan3
  */
 //!\cond
 template <typename stream_type, typename value_type>
-concept ostream_concept = std::is_base_of_v<std::ios_base, stream_type> &&
+concept ostream_concept = std::is_base_of_v<std::ios_base, std::remove_reference_t<stream_type>> &&
                                requires (stream_type & os, value_type & val)
 {
-    typename stream_type::char_type;
-    typename stream_type::traits_type;
-    typename stream_type::int_type;
-    typename stream_type::pos_type;
-    typename stream_type::off_type;
-
-    { os << val } -> std::basic_ostream<typename stream_type::char_type, typename stream_type::traits_type> &;
+    typename std::remove_reference_t<stream_type>::char_type;
+    typename std::remove_reference_t<stream_type>::traits_type;
+    typename std::remove_reference_t<stream_type>::int_type;
+    typename std::remove_reference_t<stream_type>::pos_type;
+    typename std::remove_reference_t<stream_type>::off_type;
+
+    { os << val } -> std::basic_ostream<typename std::remove_reference_t<stream_type>::char_type,
+                                        typename std::remove_reference_t<stream_type>::traits_type> &;
 };
+
+template <typename stream_type>
+concept ostream_concept2 = requires { typename std::remove_reference_t<stream_type>::char_type; } &&
```

Comment:
Hm can't you default the second template parameter instead of a second concept?:
```cpp
template <typename stream_type, typename value_type = typename std::remove_reference_t<stream_type>::char_type>
concept ostream_concept = std::is_base_of_v<std::ios_base, std::remove_reference_t<stream_type>> &&
                               requires (stream_type & os, value_type & val)
{
     // ...
};
```
EDIT: if we just assume char_type is always a mamber :grin:

Reply:
I tried, but GCC still segfaults if concepts have defaulted template parameters. I discussed this briefly with @rrahn and we might change those concept names altogether so I just used `ostream_concept2` as a placeholder for now.

Reply:
Oh, this mess with the default parameters in concept declarations 🤦‍♂️